python-package/xgboost/spark/model.py

```diff
@@ -222,11 +200,11 @@ def saveImpl(self, path):
         """
         xgb_model = self.instance._xgb_sklearn_model
         _SparkXGBSharedReadWrite.saveMetadata(self.instance, path, self.sc, self.logger)
-        model_save_path = os.path.join(path, "model.json")
-        ser_xgb_model = serialize_xgb_model(xgb_model)
-        _get_spark_session().createDataFrame(
-            [(ser_xgb_model,)], ["xgb_sklearn_model"]
-        ).write.parquet(model_save_path)
+        model_save_path = os.path.join(path, "model")
+        booster = xgb_model.get_booster().save_raw("json").decode("utf-8")
+        _get_spark_session().sparkContext.parallelize([booster], 1).saveAsTextFile(
+            model_save_path
```

Comment:
Interesting idea, but how to control the saved file name ?

Reply:
Does the `booster` string contain "\n" character ?

Reply:
It is always part-00000

Reply:
Let's document the file name "part-00000" is the model json file.

and pls add a test to ensure the model json file does not contain `\n` character and document the reason.

Reply:
Just checked the code, the file name is defined by https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala#L225.

``` scala
  override def initWriter(taskContext: NewTaskAttemptContext, splitId: Int): Unit = {
    val numfmt = NumberFormat.getInstance(Locale.US)
    numfmt.setMinimumIntegerDigits(5)
    numfmt.setGroupingUsed(false)

    val outputName = "part-" + numfmt.format(splitId)
    val path = FileOutputFormat.getOutputPath(getConf)
    val fs: FileSystem = {
      if (path != null) {
        path.getFileSystem(getConf)
      } else {
        // scalastyle:off FileSystemGet
        FileSystem.get(getConf)
        // scalastyle:on FileSystemGet
      }
    }
...
```

 here the **splitId** is the TaskContext.partitionId(). In our case, there is only 1 partition, so the file name is "part-00000"

Reply:
yes I know that. My point is can we customize the file name to make it more user-friendly.
Not a must though.

Reply:
that's the internal behavior of pyspark, not sure if it's a good idea to rely on it.

Reply:
Yeah, If you guys insist, I can use the FileSystem java API to achieve it by py4j.

Reply:
> Yeah, If you guys insist, I can use the FileSystem java API to achieve it by py4j.

No need to do that, it makes code hard to maintain, your current code is fine.